schemas/com.snowplowanalytics.snowplow.ecommerce/promotion/jsonschema/1-0-0

```diff
@@ -0,0 +1,62 @@
+{
+    "$schema": "http://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0#",
+    "description": "Schema for a promotion entity in Ecommerce",
+    "self": {
+        "vendor": "com.snowplowanalytics.snowplow.ecommerce",
+        "name": "promotion",
+        "format": "jsonschema",
+        "version": "1-0-0"
+    },
+    "type": "object",
+    "properties": {
+        "id": {
+            "description" : "The ID of the promotion.",
+            "type": "string",
+            "maxLength": 4096
+        },
+        "name": {
+            "description" : "The name of the promotion.",
+            "type": [ "string", "null" ],
+            "maxLength": 256
+        },
+        "product_ids" : {
+            "description" : "Array of SKUs or product IDs showcased in the promotion.",
+            "type" : [ "array", "null" ],
```

Comment:
Just so you know, the way we handle arrays in redshift is non-ideal.  The `product_ids` column will be a VARCHAR column with max length 65535.  We serialize the array of strings as a JSON array and load it into the VARCHAR column.

If the serialized array exceeds 65535 characters, then the string gets truncated.  So it's possible the column contains a value like this (although longer):

```
["Lorem", "ipsum", "d
```

Having said that, load of Iglu Central schemas contain array fields already.  So I don't suggest you change anything.  Just be aware of it when it comes to data modelling.

Reply:
@emielver Are we ok with that ?

Reply:
I think that should be fine, if anyone does every encounter this issue we could do some specific string processing on it for redshift, but we don't do anything for other columns at the moment so we should be okay.